resources/uco_design_document.md

```diff
@@ -71,24 +71,56 @@ To achieve its targeted objectives UCO has chosen to pursue an ontological appro
+In the UCO RDFS/OWL/SHACL ontology, all classes are defined as subclasses of the `UcoThing` class to provide a simple scoping of UCO defined "things".
+Within this scope, classes are defined for any relevant domain concept (domain objects) as well as for any structured concept characterizing some aspect of a domain concept (non-domain objects). To provide clear delineation between these, two disjoint subclasses of `UcoThing` are defined: `UcoObject` (for domain objects) and `UcoInherentCharacterizationThing` (for non-domain objects).
+
+`UcoInherentCharacterizationThing` classes inhere in `UcoObject` classes; this implies that for a `UcoInherentCharacterizationThing` concept to exist, it is dependent on the existence of the `UcoObject` concept that bears the `UcoInherentCharacterizationThing` characterizing it. For example, when destroying a red car, the car as bearer for the red color is removed and with it, its red color disappears. Note that the reverse is not true; `UcoObject` instances are not existentially dependent on `UcoInherentCharacterizationThing` instances, and, thus, cannot inhere in them. Note further that, although the example suggests that `UcoInherentCharacterizationThing` instances are compulsary for `UcoObject` concepts, this is not the case.
+
+Domain concept classes (e.g., File, Action, Identity, Location, Device, etc.) are defined as subclasses of the `UcoObject` class. Non-domain concept classes characterize domain concept classes and are defined as subclasses of the `UcoInherentCharacterizationThing` class. Domain concept classes represent the things whereas non-domain concept classes represent the thing's characteristics. The disjointness between them follows from the fact that the thing can never be the same as its characteristics.
 
-Domain concept classes (e.g., File, Action, Identity, Location, Device, etc.) are defined as subclasses of the UcoObject class. `Facet` classes characterize a particular pattern of properties that potentially apply for more than one domain class; a color, weight, an address and alike (described in https://github.com/ucoProject/ucoproject.github.io/pull/5 below) represent characteristics that apply not only for cars, but also for houses, persons, books and what have you.
+Individual instances of `UcoInherentCharacterizationThing` may appear in the fully granular RDF graph but only individual instances of `UcoObject` would appear in the domain graph. 
+In the domain graph, any individual instances of `UcoInherentCharacterizationThing` would be considered internal characteristic properties of the individual instances of `UcoObject` they are associated with.
+This means that any object properties in a UCO domain graph that do not have a range that is `UcoObject` or one of its subclasses MUST have a range that is `UcoInherentCharacterizationThing` or one of its subclasses.

```

Comment:
> Individual instances of `UcoInherentCharacterizationThing` may appear in the fully granular RDF graph but only individual instances of `UcoObject` would appear in the domain graph. In the domain graph, any individual instances of `UcoInherentCharacterizationThing` would be considered internal characteristic properties of the individual instances of `UcoObject` they are associated with. This means that any object properties in a UCO domain graph that do not have a range that is `UcoObject` or one of its subclasses MUST have a range that is `UcoInherentCharacterizationThing` or one of its subclasses.

I do not understand this, even after reading the edits below to the `Facet` section.

This reads like a significant imposition on domain graphs.  Up to this point, they've been illustrated, but not demonstrated with an example in a concrete syntax.  It looks to me like if a "full RDF graph" were done in Turtle, a "domain graph" could also be done in Turtle, but cutting away all triples involving properties "outside" that domain.  To implement this example, I could define a "email gossip domain" that only cares about these classes and properties:

* `observable:EmailAddress`
* `observable:EmailMessage`
* `observable:to`
* `observable:from`
* `observable:cc`

If domain graphs can't contain `UcoInherentCharacterizationThing`, then I can't use those three properties because they are in `EmailMessageFacet`, not `EmailMessage`, so they can't attach to anything.  The same issue applies to the `observable:sections` property in the next paragraph - that is on `WindowsPEBinaryFileFacet`, not the `UcoObject` subclass `WindowsPEBinaryFile`.